toolchain/check/import_cpp.cpp

```diff
@@ -1086,22 +1187,54 @@ static auto MakeParamPatternsBlockId(Context& context, SemIR::LocId loc_id,
   return context.inst_blocks().Add(params);
 }
 
-// Returns the return type of the given function declaration. In case of an
-// unsupported return type, it produces a diagnostic and returns
-// `SemIR::ErrorInst::InstId`.
+// Returns the return `TypeExpr` of the given function declaration. In case of
+// an unsupported return type, returns `SemIR::ErrorInst::InstId`. Constructors
+// are treated as returning a class instance.
 // TODO: Support more return types.
-static auto GetReturnType(Context& context, SemIR::LocId loc_id,
-                          const clang::FunctionDecl* clang_decl)
-    -> SemIR::InstId {
+static auto GetReturnTypeExpr(Context& context, SemIR::LocId loc_id,
+                              clang::FunctionDecl* clang_decl) -> TypeExpr {
   clang::QualType ret_type = clang_decl->getReturnType();
-  if (ret_type->isVoidType()) {
-    return SemIR::InstId::None;
+  if (!ret_type->isVoidType()) {
+    TypeExpr mapped_type = MapType(context, loc_id, ret_type);
+    if (!mapped_type.inst_id.has_value()) {
+      return {.inst_id = SemIR::ErrorInst::TypeInstId,
+              .type_id = SemIR::ErrorInst::TypeId};
+    }
+    return mapped_type;
+  }
+
+  if (!isa<clang::CXXConstructorDecl>(clang_decl)) {
```

Comment:
```suggestion
  if (!clang::isa<clang::CXXConstructorDecl>(clang_decl)) {
```

This clearly isn't required, but should we have it anyways? (note, I'm partly commenting because it's there elsewhere, I'm not looking into why it's optional)

Reply:
Done.
I've actually looked into whether I should use `isa`, `clang::isa` or `llvm::isa`.
While it seems like `clang::isa` makes the most sense for Clang classes, and that's what I've been using before, I also like to be consistent.
I've noticed `isa` is used #5796, so I was thinking that if we just need to choose one, we might as well choose the shorter version.

Reply:
I'll check with @zygoloid what his preference is.

Reply:
Dropping the prefix: https://github.com/carbon-language/carbon-lang/pull/5909

Reply:
Done.